lib/view_component/instrumentation.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require "active_support/notifications"
+
+module ViewComponent # :nodoc:
+  module Instrumentation
+    class << self
+      def included(mod)
+        mod.prepend(self)
+      end
+
+      def enabled?
+        Rails.application.config.view_component.instrumentation_enabled
+      end
+    end
+
+    def render_in(view_context, &block)
+      return super(view_context, &block) unless Instrumentation.enabled?
```

Comment:
Is this necessary? As I see, we use this check in tests.

Maybe, we can instead manually include the module in the view component class under test?

Reply:
Sure we can, but this also helps with hot reloading during development: prepended module remains after reload, so I decided to leave this line. But yeah, it's kinda ugly.

Reply:
> hot reloading during development

Actually, I think, generally, any updates to `config/*.rb` are not handled by `reload!`; so, we don't need to care about it. We only extend the core functionality once on boot.

Reply:
I agree, I think we could remove this check. 👍